### PR TITLE
[17.09] Add ICM datatype for glimmer

### PR DESCRIPTION
**Pull request by abretaud**
Hey,
Just a new little datatype needed for the update of glimmer tool in https://github.com/bgruening/galaxytools/pull/660

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1792,6 +1792,32 @@ def sniff(self, filename):
...
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary ICM (interpolated context model) file"
+            dataset.blurb = nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def sniff(self, dataset):
+        line = open(dataset).readline()
```
Comment (bgruening) [inline lib/galaxy/datatypes/binary.py]: Can we restrict how many characters we read here with `.read(100)` or so?

**Comment by bgruening** [comment]
Thanks @abretaud! Because this is a backport to a stable release we need to have two +1.
I'm not sure this is currently allowed, as we can probably not count it as bugfix. But let's see what others think.

**Comment by abretaud** [comment]
Ok we'll see, no problem to retarget if needed, I was maybe a little too optimistic!

**Comment by bgruening** [comment]
@mvdbeek could you look at it as well, please?

**Pull request merged**